sql/core/src/main/java/org/apache/spark/sql/sources/v2/Table.java

```diff
@@ -45,4 +44,9 @@
    * empty schema can be returned here.
    */
   StructType schema();
+
+  /**
+   * Returns the set of capabilities for this table.
+   */
+  Set<TableCapability> capabilities();
```

Comment:
I don't think we will have tons of capabilities, maybe `Array` is good enough? `Array` is also more java/scala friendly.

Reply:
I don't think it's a good idea to use an array when the storage should be a set, just because it is necessary to call `asJava` when returning it.